impc_prod_tracker/src/main/java/uk/ac/ebi/impc_prod_tracker/conf/security/PublicKeyProvider.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *******************************************************************************/
+package uk.ac.ebi.impc_prod_tracker.conf.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.security.PublicKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+
+/**
+ * Class in charge to provide a public key to be used to verify signatures in tokens
+ * provided by an authentication service.
+ * @author Mauricio Martinez
+ */
+@Component
+public class PublicKeyProvider
+{
+    @Autowired
```

Comment:
Prefer constructor injection instead of Autowired.

Reply:
Done.